Prepare v1.4.3 (#90)
kensanata/mastodon-archive: README.md

<!-- markdown-toc end -->

# Installation
There are multiple alternative ways to install `mastodon-archive` on your machine:

## Linux Packages
There are now packages available for Debian (`*.deb`) and Redhat (`*.rpm`) based
systems. They are not in the standard repositories, though – but for now can be
installed from [IzzySoft's Repositories](https://apt.izzysoft.de/). There you

> also find instructions on how to include them with your

Reviewer: ```suggestion
also find instructions on how to include them.
```

Reviewer: I know this sounds fine when read aloud, but visually I think paragraphs should always end with punctuation. I guess I'm taking my new day job as tech writer a bit too serious. :laughing:

Author: Punctuation, hm? May we double it then?

    also find instructions on how to include them:

Reviewer: Work for me.

Author: Done then (comes with the next push).

pip3 install mastodon-archive
```

## Manually install the latest development code

> You can always clone the repository and run `python setup.py` from within its

Reviewer: I actually never run `setup.py` directly. Do you?

```suggestion
You can always clone the repository and run `pip .` or `pip3 .` from within its
```

Author: You might have noticed I've split the Install section into 3 parts: using the package manager, using PIP – and using `setup.py`. This line is from the last sub-section.

Maybe it's too long ago that I worked with that – but when I did it last, `pip` was not on my system AFAIR. But I might be wrong. Let me check, IIRC there was a package like `python-setup-tools`… ah: `python-setuptools`, OK. What's in there? [No `pip` or `pip3`](https://packages.debian.org/buster/all/python3-setuptools/filelist), nor are they in the dependency [python3-distutils](https://packages.debian.org/buster/all/python3-distutils/filelist). [Here's an article](https://www.pythonpool.com/python-setuptools/#h-installing-via-setup-py) on the topic. Let me quote:

    python setup.py install

OK, so my line is not correct. But it seems there's `pip install <package>` – and there's `python setup.py install`? Also see [here](https://ucas.io/tutorial/python-setup/) (says to use `python setup.py install`, no word on `pip`or `pip3` at all) and– TADA! – [the official documentation](https://docs.python.org/3/install/index.html). Final proof: `python setup.py install` is correct, and it's called the "legacy version". At least valid up to Ubuntu 20.04 and derivatives. So shall I rather append `install` to the line, and keep it separate from the `pip` sub-section?

Reviewer: > You might have noticed I've split the Install section into 3 parts: using the package manager, using PIP – and using `setup.py`.

I guess what I'm arguing for is that the installation section has three parts, agreed, but that the the last part still uses pip:

1. using the package manager
2. using pip to install a released version from pypi
3. using pip to install the bleeding edge from a git checkout

Author: Hm, and where's the part for those who do not want to use PIP (peeps like me) then? Can we maybe agree to mentioning both in the third part, e.g.

```bash
git clone https://github.com/kensanata/mastodon-archive
cd mastodon-archive
python setup.py install  # alternatively, you can also use 'pip3 .'  or ' pip .' , depending on what your system provides
```
If you want, of course the heading can be changed from "Manually install the latest development code" to "Install the bleeding edge from a git checkout" :wink:

Background: 1) why should I have to install `pip` first if I don't need it otherwise, and do not really need it here? Keep it simple: what's not there doesn't bring in vulns etc. 2) Aww, my system has no `pip` (long explanation how to get it onto that system – all while… err… see 1) and 3) always good to have alternatives. Even to alternatives :wink:

Author: Aw, missed your first comment here. OK, alternative suggestion then:

```bash
git clone https://github.com/kensanata/mastodon-archive
cd mastodon-archive
pip .
```

```markdown
If your system does not provide `pip` or `pip3`, you can also use the legacy approach of `python setup.py install`. See
[Tool recommendations](https://packaging.python.org/en/latest/guides/tool-recommendations/#packaging-tool-recommendations)
for details.
```

But notice that your quote states (emphasis mine):

> to install a Python application and its dependencies **during deployment**

We're talking about a local install here – not about packaging, deployment and distribution :wink:

root directory:

```bash
git clone https://github.com/kensanata/mastodon-archive
cd mastodon-archive

> python setup.py

Reviewer: ```suggestion
pip .
```

Author: see above: append `install` instead?
